src/encryptsave/mod.rs

```diff
@@ -0,0 +1,175 @@
+mod ll;
+pub mod errors;
+
+
+macro_rules! tox_res {
+    ( $res:ident <- $rexpr:expr, $err:ident, $r:expr ) => {
```

Comment:
What about
```rust
unsafe {
    let mut pk = ::std::mem::uninitialized();
    tox_res!(
        err,
        ll::tox_derive_key_with_salt(
            passphrase.as_ptr(),
            passphrase.len(),
            salt.as_ptr(),
            &mut pk,
            &mut err
        )
    ).map(|r| ToxPassKey { passkey: r })
}
```
? If you need some additional conditions, you can chain them as well. I doubt if there's a need to put everything inside the macro.